modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -290,6 +293,15 @@ public static MetadataType getReturnType(Type returnType) {
       }
     }
 
+    if ((returnType.isSameType(Object.class) ||
+        returnType.isSameType(InputStream.class)) && type != null) {
```

Comment:
I would change it to `returnType.getMetadataType().instanceOf(BinaryType.class)` so you cover all binary cases